controller/users.go

```diff
@@ -181,6 +187,11 @@ func (c *UsersController) Update(ctx *app.UpdateUsersContext) error {
 
 		updatedUserName := ctx.Payload.Data.Attributes.Username
 		if updatedUserName != nil && *updatedUserName != identity.Username {
+			isValid := isUsernameValid(*updatedUserName)
+			if !isValid {
+				jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrInvalidRequest(fmt.Sprintf("invalid value assigned to username for identity with id %s and user with id %s", identity.ID, identity.UserID.UUID)))
+				return ctx.BadRequest(jerrors)
```

Comment:
what about adding a `NewBadRequestError(msg string)` function in `errors/errors.go` and call something like this instead: 
````go
return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError(fmt.Sprintf("invalid value assigned to username for identity with id %s and user with id %s", identity.ID, identity.UserID.UUID)))
````